swift/apple/FirezoneKit/Sources/FirezoneKit/Views/MenuBar.swift

```diff
@@ -32,60 +33,63 @@
     private var connectingAnimationImageIndex: Int = 0
     private var connectingAnimationTimer: Timer?
 
-    private var appStore: AppStore
+    private var tunnelStore: TunnelStore
     private var settingsViewModel: SettingsViewModel
-    private var loginStatus: AuthStore.LoginStatus = .signedOut
-    private var tunnelStatus: NEVPNStatus = .invalid
     private var logger: AppLogger
 
-    public init(appStore: AppStore) {
+    public init(tunnelStore: TunnelStore, settingsViewModel: SettingsViewModel, logger: AppLogger) {
       self.statusItem = NSStatusBar.system.statusItem(withLength: NSStatusItem.variableLength)
-
-      self.appStore = appStore
-      self.settingsViewModel = appStore.settingsViewModel
-
-      self.logger = appStore.logger
-
+      self.tunnelStore = tunnelStore
+      self.settingsViewModel = settingsViewModel
+      self.logger = logger
       super.init()
-      createMenu()
-      setupObservers()
 
       if let button = statusItem.button {
         button.image = signedOutIcon
       }
 
-      updateStatusItemIcon()
+      createMenu()
+      setupObservers()
     }
 
     private func setupObservers() {
-      appStore.authStore.$loginStatus
-        .receive(on: mainQueue)
-        .sink { [weak self] loginStatus in
-          self?.loginStatus = loginStatus
-          self?.updateStatusItemIcon()
-          self?.handleLoginOrTunnelStatusChanged()
-        }
-        .store(in: &cancellables)
-
-      appStore.tunnelStore.$status
+      tunnelStore.$status
         .receive(on: mainQueue)
         .sink { [weak self] status in
-          self?.tunnelStatus = status
-          self?.updateStatusItemIcon()
-          self?.handleLoginOrTunnelStatusChanged()
-          self?.handleMenuVisibilityOrStatusChanged()
+          guard let self = self else { return }
+          self.logger.log("\(#function): status changed: \(status)")
+          self.updateStatusItemIcon()
+          self.handleTunnelStatusChanged()
+          self.handleMenuVisibilityOrStatusChanged()
         }
         .store(in: &cancellables)
 
-      appStore.tunnelStore.$resources
+      tunnelStore.$resourceListJSON
         .receive(on: mainQueue)
-        .sink { [weak self] resources in
+        .sink { [weak self] json in
           guard let self = self else { return }
-          self.setOrderedResources(resources.orderedResources)
+
+          let newResources = decodeResources(json)
+
+          // Update menu in-place for smooth sexy easy on the eyes UI updates
```

Comment:
```suggestion
          // Update menu in-place for smooth, silky, easy-on-the-eyes UI updates
```

Better safe than sorry.

So does that mean on macOS it can update the menu with an animation even while it's open?

Reply:
Lol, sorry, you're right. I wrote that late at night 🙈